There are issues being reported:

```
[ERROR] One or more dependencies were identified with vulnerabilities: 
[ERROR] 
[ERROR] el-api-6.0.53.jar: CVE-2012-5568, CVE-2020-8022, CVE-2013-4444, CVE-2013-2185
[ERROR] jasper-el-6.0.53.jar: CVE-2012-5568, CVE-2020-8022, CVE-2013-4444, CVE-2013-2185
[ERROR] 
```

Newer version are available at https://search.maven.org/artifact/org.apache.tomcat/tomcat-jasper-el and https://search.maven.org/artifact/org.apache.tomcat/tomcat-el-api


Update: Should be fixed by #495

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,3 +3,4 @@ language: java
 jdk: oraclejdk8
 # fixing  broken build, see https://travis-ci.community/t/error-installing-oraclejdk8-expected-feature-release-number-in-range-of-9-to-14-but-got-8/3766/7
 dist: trusty
+script: mvn clean verify -B -Pcheck-dependencies

--- a/accesscontroltool-bundle/maximum-aem.bndrun
+++ b/accesscontroltool-bundle/maximum-aem.bndrun
@@ -13,7 +13,5 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-# workaround bug https://github.com/bndtools/bnd/issues/4189
--runrequires: osgi.identity;filter:='(osgi.identity=biz.netcentric.cq.tools.accesscontroltool.bundle)
 -include ../target-osgi-environment/maximum-environment/maximum-aem.bndrun
 

--- a/accesscontroltool-bundle/minimum-aem.bndrun
+++ b/accesscontroltool-bundle/minimum-aem.bndrun
@@ -13,7 +13,5 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-# workaround bug https://github.com/bndtools/bnd/issues/4189
--runrequires: osgi.identity;filter:='(osgi.identity=biz.netcentric.cq.tools.accesscontroltool.bundle)
 -include ../target-osgi-environment/minimum-environment/minimum-aem.bndrun
 

--- a/accesscontroltool-bundle/pom.xml
+++ b/accesscontroltool-bundle/pom.xml
@@ -123,11 +123,9 @@
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-api</artifactId>
         </dependency>
-
         <dependency>
             <groupId>org.yaml</groupId>
             <artifactId>snakeyaml</artifactId>
-            <version>1.27</version>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
@@ -155,13 +153,12 @@
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>oak-core</artifactId>
-            <version>1.2.7</version>
+            <optional>true</optional>
         </dependency>
 
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>oak-auth-external</artifactId>
-            <version>1.2.7</version>
             <optional>true</optional>
         </dependency>
 
@@ -173,12 +170,10 @@
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.servlets.annotations</artifactId>
-            <version>1.2.4</version>
         </dependency>
         <dependency>
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.webconsole</artifactId>
-            <version>4.2.16</version>
             <optional>true</optional>
         </dependency>
 

--- a/accesscontroltool-startuphook-bundle/pom.xml
+++ b/accesscontroltool-startuphook-bundle/pom.xml
@@ -29,7 +29,12 @@
             <artifactId>accesscontroltool-bundle</artifactId>
             <version>${project.version}</version>
         </dependency>
-        
+        <!-- only used for constant value (no run time dependency)  -->
+        <dependency>
+            <groupId>org.apache.jackrabbit</groupId>
+            <artifactId>oak-core</artifactId>
+            <optional>true</optional>
+        </dependency>
         <!-- OSGi annotations (build time, but only use those compliant with R6) -->
         <dependency>
             <groupId>org.osgi</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -144,6 +144,11 @@
                 <artifactId>org.osgi.annotation.bundle</artifactId>
                 <version>1.0.0</version>
             </dependency>
+            <dependency>
+                <groupId>org.apache.sling</groupId>
+                <artifactId>org.apache.sling.servlets.annotations</artifactId>
+                <version>1.2.4</version>
+            </dependency>
             <!-- Apache Felix 4.6.1 (shipped in AEM 6.1) is fully compliant with OSGi Core R6 -->
             <dependency>
                 <groupId>org.osgi</groupId>
@@ -157,18 +162,21 @@
                 <groupId>org.osgi</groupId>
                 <artifactId>org.osgi.service.cm</artifactId>
                 <version>1.5.0</version>
+                <scope>provided</scope>
             </dependency>
             <!-- Event Admin 1.3.1 (https://osgi.org/javadoc/r6/cmpn/org/osgi/service/event/package-summary.html) -->
             <dependency>
                 <groupId>org.osgi</groupId>
                 <artifactId>org.osgi.service.event</artifactId>
                 <version>1.3.1</version>
+                <scope>provided</scope>
             </dependency>
             <!-- URL Stream Handlers 1.0.0 (https://osgi.org/javadoc/r6/core/org/osgi/service/url/package-summary.html) -->
             <dependency>
                 <groupId>org.osgi</groupId>
                 <artifactId>org.osgi.service.url</artifactId>
                 <version>1.0.0</version>
+                <scope>provided</scope>
             </dependency>
             <!-- only constants are referenced from there which are inlined during compilation anyways, therefore no runtime dependency -->
             <dependency>
@@ -255,7 +263,6 @@
                 <groupId>org.yaml</groupId>
                 <artifactId>snakeyaml</artifactId>
                 <version>1.27</version>
-                <type>bundle</type>
             </dependency>
             <!-- due to https://bugs.openjdk.java.net/browse/JDK-8231581 OOTB JRE is not sufficient -->
             <dependency>
@@ -276,7 +283,27 @@
                 <scope>provided</scope>
                 <optional>true</optional>
             </dependency>
-
+            <dependency>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>org.apache.felix.webconsole</artifactId>
+                <version>4.2.16</version>
+                <scope>provided</scope>
+                <optional>true</optional>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.jackrabbit</groupId>
+                <artifactId>oak-core</artifactId>
+                <version>1.2.7</version>
+                <scope>provided</scope>
+                <optional>true</optional>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.jackrabbit</groupId>
+                <artifactId>oak-auth-external</artifactId>
+                <version>1.2.7</version>
+                <scope>provided</scope>
+                <optional>true</optional>
+            </dependency>
             <!-- START: Optional AEM dependencies -->
             <dependency>
                 <groupId>com.adobe.granite</groupId>
@@ -624,6 +651,35 @@
             </build>
         </profile>
 
+        <profile>
+            <id>dependency-check</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.owasp</groupId>
+                        <artifactId>dependency-check-maven</artifactId>
+                        <version>6.0.2</version>
+                        <executions>
+                            <execution>
+                                <id>check-dependencies-for-vulnerabilities</id>
+                                <phase>validate</phase>
+                                <goals>
+                                    <goal>check</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <cveValidForHours>24</cveValidForHours>
+                            <failBuildOnAnyVulnerability>true</failBuildOnAnyVulnerability>
+                            <skipProvidedScope>true</skipProvidedScope>
+                            <skipSystemScope>true</skipSystemScope>
+                            <retireJsAnalyzerEnabled>false</retireJsAnalyzerEnabled>
+                            <suppressionFile>./suppression.xml</suppressionFile>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
     </profiles>
 
 </project>

--- a/suppression.xml
+++ b/suppression.xml
@@ -1,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.3.xsd">
+  <suppress>
+    <notes><![CDATA[
+   https://nvd.nist.gov/vuln/detail/CVE-2020-8022 marks it as vulnerable as it has version ranges for both tomcat 8 and 9.
+   ]]></notes>
+    <packageUrl regex="true">^pkg:maven/org\.apache\.tomcat/tomcat\-jasper\-el@.*$</packageUrl>
+    <cve>CVE-2020-8022</cve>
+  </suppress>
+  <suppress>
+    <notes><![CDATA[
+   https://nvd.nist.gov/vuln/detail/CVE-2016-0956 and https://nvd.nist.gov/vuln/detail/CVE-2013-4390 have very broad cpes
+   and match e.g. the sling servlet annotations
+   ]]></notes>
+    <cve>CVE-2016-0956</cve>
+    <cve>CVE-2013-4390</cve>
+  </suppress>
+</suppressions>

--- a/target-osgi-environment/maximum-environment/pom.xml
+++ b/target-osgi-environment/maximum-environment/pom.xml
@@ -67,149 +67,178 @@
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.framework</artifactId>
             <version>6.0.2</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.scr</artifactId>
             <version>2.1.16</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.configadmin</artifactId>
             <version>1.9.12</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.http.servlet-api</artifactId>
             <version>1.1.2</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
             <version>1.7.25</version>
+            <scope>provided</scope>
         </dependency>
         <!-- transitively required by slf4j-api -->
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-simple</artifactId>
             <version>1.7.25</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
             <version>15.0</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>commons-io</groupId>
             <artifactId>commons-io</artifactId>
             <version>2.6</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>commons-collections</groupId>
             <artifactId>commons-collections</artifactId>
             <version>3.2.2</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>commons-codec</groupId>
             <artifactId>commons-codec</artifactId>
             <version>1.14</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.day.commons.osgi.wrapper</groupId>
             <artifactId>com.day.commons.osgi.wrapper.commons-lang2</artifactId>
             <version>2.5-0001</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
             <version>3.3.2</version>
+            <scope>provided</scope>
         </dependency>
         <!--
             <dependency>
             <groupId>org.daisy.libs</groupId>
             <artifactId>commons-httpclient</artifactId>
             <version>3.1.0</version>
+            <scope>provided</scope>
             </dependency> -->
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.api</artifactId>
             <version>2.20.0</version>
+            <scope>provided</scope>
         </dependency>
         <!--
             <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.log</artifactId>
             <version>5.0.0</version>
+            <scope>provided</scope>
             </dependency>
             <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.classloader</artifactId>
             <version>1.3.8</version>
+            <scope>provided</scope>
             </dependency> -->
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.json</artifactId>
             <version>2.0.20</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.serviceusermapper</artifactId>
             <version>1.4.6</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>javax.jcr</groupId>
             <artifactId>jcr</artifactId>
             <version>2.0</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>oak-jackrabbit-api</artifactId>
             <version>${oak.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-webdav</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-jcr-commons</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-spi-commons</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-spi</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.jcr.base</artifactId>
             <version>3.0.4</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.settings</artifactId>
             <version>1.3.10</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.threads</artifactId>
             <version>3.2.18</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.scheduler</artifactId>
             <version>2.7.0</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.metrics</artifactId>
             <version>1.2.6</version>
+            <scope>provided</scope>
         </dependency>
         <!-- close to com.adobe.granite.dropwizard.metrics -->
         <dependency>
@@ -222,12 +251,14 @@
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.jcr.oak.server</artifactId>
             <version>1.2.4</version> <!-- uses Oak 1.3.7 -->
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.adobe.aem</groupId>
             <artifactId>uber-jar</artifactId>
             <classifier>apis</classifier>
             <version>6.5.5</version>
+            <scope>provided</scope>
         </dependency>
     </dependencies>
 

--- a/target-osgi-environment/minimum-environment/pom.xml
+++ b/target-osgi-environment/minimum-environment/pom.xml
@@ -66,146 +66,180 @@
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.framework</artifactId>
             <version>4.6.1</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.scr</artifactId>
             <version>2.0.0</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.configadmin</artifactId>
             <version>1.8.0</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.apache.felix.eventadmin</artifactId>
+            <version>1.4.2</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.felix</groupId>
             <artifactId>org.apache.felix.http.servlet-api</artifactId>
             <version>1.1.0</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
             <version>1.7.6</version>
+            <scope>provided</scope>
         </dependency>
         <!-- transitively required by slf4j-api -->
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-simple</artifactId>
             <version>1.7.6</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>jcl-over-slf4j</artifactId>
             <version>1.7.6</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
             <version>15.0</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>commons-io</groupId>
             <artifactId>commons-io</artifactId>
             <version>2.4</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>commons-collections</groupId>
             <artifactId>commons-collections</artifactId>
             <version>3.2.1</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>commons-codec</groupId>
             <artifactId>commons-codec</artifactId>
             <version>1.10</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.day.commons.osgi.wrapper</groupId>
             <artifactId>com.day.commons.osgi.wrapper.commons-lang2</artifactId>
             <version>2.5-0001</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
             <version>3.3.2</version>
+            <scope>provided</scope>
         </dependency>
         <!--
             <dependency>
             <groupId>org.daisy.libs</groupId>
             <artifactId>commons-httpclient</artifactId>
             <version>3.1.0</version>
+            <scope>provided</scope>
             </dependency> -->
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.api</artifactId>
             <version>2.9.0</version>
+            <scope>provided</scope>
         </dependency>
         <!--
             <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.log</artifactId>
             <version>5.0.0</version>
+            <scope>provided</scope>
             </dependency>
             <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.classloader</artifactId>
             <version>1.3.8</version>
+            <scope>provided</scope>
             </dependency> -->
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.commons.json</artifactId>
             <version>2.0.10</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>javax.jcr</groupId>
             <artifactId>jcr</artifactId>
             <version>2.0</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-api</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-webdav</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-jcr-commons</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-spi-commons</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.jackrabbit</groupId>
             <artifactId>jackrabbit-spi</artifactId>
             <version>${jackrabbit.version}</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.jcr.base</artifactId>
             <version>3.0.0</version>
+            <scope>provided</scope>
         </dependency>
         <!-- close to com.adobe.granite.repository 1.1.67, shipping with AEM 6.1.1 -->
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.jcr.oak.server</artifactId>
             <version>1.0.0</version> <!-- uses Oak 1.3.7 -->
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.scripting.jsp</artifactId>
             <version>2.1.6</version>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.adobe.aem</groupId>
             <artifactId>uber-jar</artifactId>
             <classifier>obfuscated-apis</classifier>
             <version>6.1.0-SP1-B0001</version>
+            <scope>provided</scope>
         </dependency>
     </dependencies>
 

